[ci] Fix project-list for spring

adangel opened the pull request:
## Describe the PR

We modify (among other files) AutowiredAnnotationBeanPostProcessor.java
but only, when we intend to build the project. This leads to
the situation, that the baseline was created against the
modified file, but PRs use the unmodified file. This is because
the regression-tester does a "git reset --hard" before running
PMD.

Now we patch spring always and only skip the gradle calls.

This should fix Regression Reports like this one (from #3807):

![grafik](https://user-images.githubusercontent.com/1573684/158799884-bf91cd85-5c19-4c16-96c1-f182a34476c5.png)

pmd-test commented:
<!--
  0 Errors
  0 Warnings
  1 Message: No java rules are changed!
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">No java rules are changed!</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

adangel commented:
That's most likely the reason why manual integration test of pmd regression tester failed: 

```
#############################: test_case_3_change_in_core
  {:errors=>{:changed=>0, :new=>0, :removed=>0, :base_total=>37, :patch_total=>37}, :violations=>{:changed=>86, :new=>0, :removed=>1, :base_total=>823311, :patch_total=>823310}, :configerrors=>{:changed=>0, :new=>0, :removed=>0, :base_total=>3, :patch_total=>3}}
  #############################
  F
  ===============================================================================
  Failure: test_case_3_change_in_core(ManualIntegrationTests)
  test/manual_integration_tests.rb:117:in `test_case_3_change_in_core'
       114: 
       115:     print "#############################: test_case_3_change_in_core\n" \
       116:           "#{@summary}\n#############################\n"
    => 117:     assert_equal(0, @summary[:violations][:changed], 'found changed violations')
       118:     assert_equal(0, @summary[:violations][:new], 'found new violations')
       119:     assert_equal(0, @summary[:violations][:removed], 'found removed violations')
       120:     assert_equal(0, @summary[:errors][:removed], 'found removed errors')
  found changed violations
  <0> expected but was
  <86>
```

Same numbers at least - 86 changed violations, 1 removed violation.
-> https://github.com/pmd/pmd-regression-tester/actions/runs/1915069424

adangel commented:
I can confirm, it's the same problem
